client/my-sites/site-settings/podcasting-details/index.jsx

```diff
@@ -349,7 +350,16 @@ class PodcastingDetails extends Component {
 	};
 
 	onCategoryCleared = () => {
-		this.props.updateFields( { podcasting_category_id: '0' } );
+		const { updateFields, clearDirtyFields, submitForm, siteSlug } = this.props;
+
+		updateFields( { podcasting_category_id: '0' }, () => {
+			submitForm();
+		} );
+
+		// Save changed fields and allow redirect
+		clearDirtyFields();
+
+		page.redirect( '/settings/writing/' + siteSlug );
```

Comment:
This redirect happens even if the save fails... that doesn't seem correct.

Reply:
Good catch - should be taken care of in 654dd34